terminus/geometry/bounding_box.py

```diff
@@ -0,0 +1,33 @@
+from geometry.point import Point
+
+
+class BoundingBox(object):
+
+    def _normalize(self):
+        new_origin = Point(min(self.origin.x, self.corner.x),
+                           min(self.origin.y, self.corner.y))
+        new_corner = Point(max(self.origin.x, self.corner.x),
+                           max(self.origin.y, self.corner.y))
+        self.origin = new_origin
+        self.corner = new_corner
+
+    def __init__(self, origin, corner):
+        self.origin = origin
+        self.corner = corner
+        self._normalize()
+
+    def __eq__(self, other):
+        return self.origin == other.origin and self.corner == other.corner
+
+    def merge(self, other):
+        merge_origin = Point(min(self.origin.x, other.origin.x),
+                             min(self.origin.y, other.origin.y))
+        merge_corner = Point(max(self.corner.x, other.corner.x),
+                             max(self.corner.y, other.corner.y))
+        return BoundingBox(merge_origin, merge_corner)
+
+    def merge_multiple_boundingboxes(self, box_list):
```

Comment:
While the idea is ok, it seems weird that you are asking a bounding box to merge other bounding boxes but not to participate at all in the process or final result. Like you can do:

```
boxes = [BoundingBox(Point(4, 4), Point(5, 5)),
              BoundingBox(Point(4.5, 4.5), Point(6, 6))]
 BoundingBox(Point(0, 0), Point(1, 1)).merge_multiple_boundingboxes(boxes)
```
and the resulting bounding box has nothing to do with `BoundingBox(Point(0, 0), Point(1, 1))`

What you want instead is to be able to create a new bounding box from a collection of bounding boxes. To do that you need as a class method that acts as an alternative constructor so you can do things like:

```
boxes = [BoundingBox(Point(4, 4), Point(5, 5)),
              BoundingBox(Point(4.5, 4.5), Point(6, 6))]
BoundingBox.from_boxes(boxes)
```

the implementation would be quite similar to yours:

```
@classmethod
def from_boxes(cls, box_list):
       whole_merge = box_list[0]
        for i in range(1, len(box_list)):
            whole_merge = whole_merge.merge(box_list[i])
        return whole_merge
```